webui/react/src/pages/ExperimentDetails/ExperimentCheckpoints.tsx

```diff
@@ -122,16 +122,39 @@ const ExperimentCheckpoints: React.FC<Props> = ({ experiment, pageRef }: Props)
     [openModalCheckpointRegister],
   );
 
+  const handleDelete = useCallback((checkpoints: string[]) => {
+    readStream(
+      detApi.Checkpoint.deleteCheckpoints({
+        checkpointUuids: checkpoints,
+      }),
+    );
+  }, []);
+
   const handleDeleteCheckpoint = useCallback(
     (checkpoints: string[]) => {
-      openModalCheckpointDelete({ checkpoints });
+      const content = `Are you sure you want to request checkpoint deletion for ${
+        checkpoints.length
+      }
+      ${pluralizer(
+        checkpoints.length,
+        'checkpoint',
+      )}? This action may complete or fail without further notification.`;
+
+      confirm({
+        content,
+        danger: true,
+        okText: 'Request Delete',
+        onConfirm: () => handleDelete(checkpoints),
+        onError: handleError,
+        title: 'Confirm Checkpoint Deletion',
+      });
```

Comment:
That would make sense. The reason I included it in this PR is that the old modal used an icon in the header that broke after the refactor in a way that would be annoying to fix, especially since it was in code that we would want to remove eventually. Since this work is already done I'd like to keep it in, but I'll keep that in mind for the future.